core/src/main/scala/kafka/server/metadata/KRaftMetadataCache.scala

```diff
@@ -341,10 +341,10 @@ class KRaftMetadataCache(
     Option(_currentImage.cluster.broker(brokerId)).count(_.inControlledShutdown) == 1
   }
 
-  private def getAliveBrokers(image: MetadataImage): util.List[BrokerMetadata] = {
+  private def getAliveBrokers(image: MetadataImage): util.List[UsableBroker] = {
```

Comment:
we should inline this method. for example:
```scala
  private def getRandomAliveBroker(image: MetadataImage): util.Optional[Integer] = {
    val aliveBrokers = image.cluster().brokers().values().stream()
      .filter(Predicate.not(_.fenced))
      .map(_.id()).toList
    if (aliveBrokers.isEmpty) {
      util.Optional.empty()
    } else {
      util.Optional.of(aliveBrokers.get(ThreadLocalRandom.current().nextInt(aliveBrokers.size)))
    }
  }
```

Reply:
Address it. Thanks.